freqtrade/optimize/backtesting.py

```diff
@@ -275,8 +275,30 @@ def load_bt_data_detail(self) -> None:
                 if pair not in self.exchange._leverage_tiers:
                     unavailable_pairs.append(pair)
                     continue
-                self.futures_data[pair] = funding_rates_dict[pair].merge(
-                    mark_rates_dict[pair], on='date', how="inner", suffixes=["_fund", "_mark"])
+
+                if (pair in mark_rates_dict
+                        and len(funding_rates_dict[pair]) == 0
+                        and "futures_funding_rate" in self.config):
+                    mark_rates_dict[pair]["open_fund"] = self.config.get('futures_funding_rate')
+                    mark_rates_dict[pair].rename(
+                        columns={'open': 'open_mark',
+                                 'close': 'close_mark',
+                                 'high': 'high_mark',
+                                 'low': 'low_mark',
+                                 'volume': 'volume_mark'},
+                        inplace=True)
+
+                    self.futures_data[pair] = mark_rates_dict[pair]
+                else:
+                    if "futures_funding_rate" in self.config:
+                        self.futures_data[pair] = mark_rates_dict[pair].merge(
+                            funding_rates_dict[pair], on='date',
+                            how="outer", suffixes=["_mark", "_fund"])['open_fund'].fillna(
+                                self.config.get('futures_funding_rate'))
```

Comment:
This is ... actually wrong.

it's selecting a series - and then assigns the series to `futures_data`.
I'm refactoring this a bit now to move this to the exchange class so all the "combine" logic is in one place - you'll see what i mean. That'll also allow tests on it - which i think is quite important (this is what identified this bug in the first place) :laughing: 

I'd appreciate if you could give a look at the changes i made - maybe have a run with "your" scenario, too - to see if this is what you're after.